hasher-matcher-actioner/hmalib/common/tests/test_failing_dataclass_serialization.py

```diff
@@ -0,0 +1,29 @@
+import typing as t
+from dataclasses import dataclass, field
+
+from hmalib.common import aws_dataclass
+
+
+@dataclass
+class NotSoSimple(aws_dataclass.HasAWSSerialization):
+    e: t.Set[float] = field(default_factory=lambda: {1.3, 4.2})
+
+
+"""
+I have no idea what's going on. If you comment out the class defined above and
+run tests, they fail with a deserialization error.
+
+My numerous attempts at figuring out exactly what is wrong have failed. The
```

Comment:
I think the fault is in python typing. I dived into the HasAWSSerialization, and there does not seem to be any reflection that would cause this.

We'll fight this another day.

Reply:
The best part of python framework hacking is trying to debug it when it fails! This is generally why I suggest avoiding python magic if possible.